packages/plugins/plugin-idfa/ios/AnalyticsReactNativePluginIdfa.swift

```diff
@@ -16,21 +16,28 @@ class AnalyticsReactNativePluginIdfa: NSObject {
     ) -> Void {
         if #available(iOS 14, *) {
             ATTrackingManager.requestTrackingAuthorization { status in
-                let idfa = status == .authorized ? ASIdentifierManager.shared().advertisingIdentifier.uuidString : self.fallbackValue
-                resolve([
-                    "adTrackingEnabled": status == .authorized,
-                    "advertisingId": idfa!,
-                    "trackingStatus": self.statusToString(status)
-                ])
+                let idfa = status == .authorized ? ASIdentifierManager.shared().advertisingIdentifier.uuidString : nil
+                if let authorizedIDFA = idfa {
```

Comment:
nit: You could refactor this to check only for status in the if then assign the value to avoid the ternary:

```swift
if status == .authorized {
  let idfa = ASIdentifierManager.shared().advertisingIdentifier.uuidString
  //...
}
//...
```

Reply:
refactored per this suggestion